### PR TITLE
get draining parameter from aux db

### PR DESCRIPTION
**Pull request by ticoann**
replace #7731 #7923

Also #8023 need to be added (won't be merged until new agent is ready)

### CODE REVIEW

**Comment by ticoann** [comment]
@amaltaro, Alan please review.

**Comment by ticoann** [comment]
test this please

File: src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py [hunk 1]
```diff
@@ -57,14 +57,76 @@ def _getDataFromMemoryCache(self, callname):
...
+        """
+        retrieve agent configuration reqmgr aux db.
+        """
+        return self._getDataFromMemoryCache('wmagentconfig/%s' % agentName)
+
+    def postWMAgentConfig(self, agentName, agentConfig):
+
+        return self["requests"].post('wmagentconfig/%s' % agentName, agentConfig)[0]['result']
+    
+    def isWMAgentDraining(self, agentName, agentDrainCondition=None):
```
Comment (amaltaro) [inline src/python/WMCore/Services/ReqMgrAux/ReqMgrAux.py]: Now that I finished reading this module. I think we should separate getters from setters. And this method is clearly a getter (by its name), but it also uploades the couch record.

Just an idea, whatever is calling `isDrainMode` should first call a `updateDrainMode` or something like that and only then call `isDrainMode`. 
Second option would be renaming this method to `updateDrainMode` and moving the final logic to `isDrainMode`.
Third option is something better that you'll probably suggest :)

**Review by amaltaro — CHANGES_REQUESTED** [review]
A few minor things to fix and possible some other minor improvements. Until those are fixed, you have my red flag :-)
And sorry for taking so long to review it!

File: bin/deploy-wmagent.sh [hunk 2]
```diff
@@ -312,6 +312,12 @@ else
 fi
 echo "Done!" && echo
 
+### Upload WMAgentConfig to AuxDB
+echo "*** Upload WMAgentConfig to AuxDB ***"
+cd $MANAGE
+./manage execute-agent wmagent-upload-config
+echo "Done!" && echo
```
Comment (ticoann) [inline bin/deploy-wmagent.sh]: Yes, you're right. I will do that.

**Comment by ticoann** [comment]
test this please

**Comment by ticoann** [comment]
test this please

File: src/python/WMCore/ReqMgr/Service/Auxiliary.py [hunk 3]
```diff
@@ -186,10 +195,41 @@ def post(self):
...
+        if subName:
+            docName = "%s_%s" % (self.name, subName)
+        else:
+            docName = self.name
+        doc = Document(docName, doc)
         result = self.reqmgr_aux_db.commitOne(doc)
         return result
 
+    @restcall(formats=[('text/plain', PrettyJSONFormat()), ('application/json', JSONFormat())])
+    def put(self, subName=None):
```
Comment (ticoann) [inline src/python/WMCore/ReqMgr/Service/Auxiliary.py]: Hmm, I didn't see previous reply for this. (I am pretty sure I did).  
The problem is there is "_"  prefix from subName so even if we have empty sting as default we still need a if else statement.

File: src/python/WMCore/ReqMgr/Service/Auxiliary.py [hunk 4]
```diff
@@ -186,10 +195,41 @@ def post(self):
...
+            propertyDict = json.loads(data)
+
+        try:
+            result = None
+            if subName:
+                docName = "%s_%s" % (self.name, subName)
+            else:
+                docName = self.name
+
+            if propertyDict:
```
Comment (ticoann) [inline src/python/WMCore/ReqMgr/Service/Auxiliary.py]: Every property values is required explicitly. So if agent is out of drain, it should have explicitly say {"AgentDrainMode": False} and current code (with this patch) will do that.

**Pull request merged**